migrations/20191230174852-projPhaseFromDocs.js

```diff
@@ -0,0 +1,418 @@
+      })
+
+      console.log("Projects with docs missing any phase field found: ", projectsWithoutDocPhase.length);
+      let docphase_cols = [{ key: 'projectId', header: 'projectId' }, { key: 'name', header: 'name' }];
+      CSV.stringify(projectsWithoutDocPhase, { header: true, columns: docphase_cols }, function (err, data) {
+        fs.writeFileSync("migrations_temp/ProjectsWithoutDocPhase.csv", data)
+      })
+
+      console.log("Projects that could be updated by documents: ", updateCandidates.length);
+      let candidate_cols = [
+        { key: 'project', header: 'projectId' },
+        { key: 'recentDoc', header: 'mostRecentDoc' },
+        { key: 'projectName', header: 'projectName'},
+        { key: 'docPhase', header: 'docPhase' },
+        { key: 'phaseList', header: 'phaseList'},
+      ];
+      CSV.stringify(updateCandidates, { header: true, columns: candidate_cols }, function (err, data) {
+        fs.writeFileSync("migrations_temp/TaggingCandidates.csv", data)
+      })
+
+      mClient.close();
+    }).catch((e) => {
+      console.log("error: ", e);
+      mClient.close()
+    });
+};
+
+function amendmentDone(doc) {
+  if (!doc.projectPhase[0] || !doc.milestone[0] || !doc.type[0]) {
+    // need all 3 to determine if done amendment
+    return false;
+  }
+  // phase: amendment, milestone: amendment, type: amendment package, then amendment is done
+  if (doc.type[0].name === "Amendment Package" && doc.milestone[0].name === "Amendment") {
+    // console.log("Found final amendment doc: ", doc._id )
+    return true;
+  } else {
+    return false;
+  }
+}
+
+// pass project
+function matchPhaseToDecision(decisionPhases, docPhaseStack, phases) {
+  let phaseToTag = '';
+  let previousPhases = [];
+  let stackCopy = [...docPhaseStack];
+
+  if (docPhaseStack.length === 1) {
+    phaseToTag = docPhaseStack.pop()
+  }
+  // todo is this getting hit?
+  while (!phaseToTag && !docPhaseStack) {
+    let phaseCandidate = docPhaseStack.pop();
+    if (decisionPhases.includes(phaseCandidate)) {
+      console.log(`Most recent phase: ${phaseCandidate}, stack: ${stackCopy} `)
+      phaseToTag = phaseCandidate;
+      previousPhases = docPhaseStack;
+    } else {
+      console.log(`phase not in Ea decision set: ${phaseCandidate}, stack: ${stackCopy} `)
+    }
+  }
+  // popped whole stack, no matching phase found, use most recent
+  if (!phaseToTag) {
+    phaseToTag = stackCopy.pop();
+    previousPhases = stackCopy;
+    // console.log("popped all stack, defaulting to most recent: ", phaseToTag );
+  }
+
+  var phaseToTagId = lookupId(phases, phaseToTag);
+  var previousPhasesIds = namesToIds(phases, previousPhases);
+  return { "currentPhase": phaseToTagId, "previousPhases": previousPhasesIds, "previousNames": previousPhases }
+}
+
+function lookupId(phases, name) {
+  // console.log("looking up: ", name);
+  var phaseId;
+  [phaseId] = phases.filter(phase => {
```

Comment:
Not sure that the square brackets are required. They generally indicate computed properties, but shouldn't be required here.

Reply:
Just some destructuring